src/pages/index.vue

```diff
@@ -53,72 +57,67 @@ const props = defineProps({
 });
 
-const countryToSelect: Ref<null | string> = ref(null);
+const checkRouteProp = (propName: PropName, candidates: string[]) => {
+    // Do case-insensitive check against route prop - add to unknown props if not found
+    if (props[propName] === "") return "";
+    const pattern = new RegExp(`^${props[propName]}$`, "i");
+    return candidates.find((i) => pattern.test(i));
+};
 
 const selectDataForRoute = async () => {
-    if (!appConfig.value) {
-        return;
-    }
+    if (!appConfig.value) return;
 
     const unknown = [];
-    const checkRouteProp = (propName: "pathogen" | "version" | "indicator" | "country", candidates: string[]) => {
-        if (!props[propName]) {
-            return "";
-        }
-
-        // Do case-insensitive check against route prop - add to unknown props if not found
-        const pattern = new RegExp(`^${props[propName]}$`, "i");
-        const result = candidates.find((i) => pattern.test(i));
-        if (!result) {
-            unknown.push(propName);
-        }
-        return result;
+    const possibleValuesForProps: Record<PropName, string[]> = {
+        pathogen: [PATHOGEN],
+        version: [VERSION],
+        indicator: Object.keys(appConfig.value.indicators),
+        country: appConfig.value.countries
     };
-
-    checkRouteProp("pathogen", [PATHOGEN]);
-    checkRouteProp("version", [VERSION]);
-    const indicator = checkRouteProp("indicator", Object.keys(appConfig.value.indicators));
-    const country = checkRouteProp("country", appConfig.value.countries);
+    Object.keys(possibleValuesForProps).forEach((prop: PropName) => {
+        if (checkRouteProp(prop, possibleValuesForProps[prop]) === undefined) {
+            unknown.push(prop);
+        }
+    });
 
     unknownProps.value = unknown;
+    if (unknownProps.value.length) return;
+
+    let path = "";
+    path += `/${props.pathogen || PATHOGEN}`;
+    path += `/${props.version || VERSION}`;
+    path += `/${props.indicator || indicatorNames.value[0]}`;
+    const currentPath = router.currentRoute.value.path;
+    if (currentPath.indexOf(path) !== 0) {
+        router.replace(path);
+    }
 
-    if (!unknownProps.value.length) {
-        if (indicator) {
-            selectedIndicator.value = indicator;
-            if (country !== selectedCountryId.value) {
-                countryToSelect.value = country;
-            }
-        } else {
-            // No indicator selected on route - default to first indicator and navigate
-            router.replace(`/${APP_BASE_ROUTE}/${indicatorNames.value[0]}`);
-        }
+    const newMapSettings: MapSettings = {
+        pathogen: checkRouteProp("pathogen", possibleValuesForProps.pathogen),
+        version: checkRouteProp("version", possibleValuesForProps.version),
+        indicator: checkRouteProp("indicator", possibleValuesForProps.indicator),
+        country: checkRouteProp("country", possibleValuesForProps.country),
```

Comment:
we call `checkRouteProp` again just to get the correct casing

Reply:
:(